### ISSUE TITLE
Incorrect current section highlighting on partial match

### ISSUE THREAD

**Issue by sapegin** [styleguidist/react-styleguidist#1237]
**Current behavior**

Several sections are highlighted when the current section contains part of a name of a section in the sidebar.

**To reproduce**

<img width="370" alt="image 2018-12-24 at 7 57 53 pm" src="https://user-images.githubusercontent.com/70067/50405538-77641200-07b6-11e9-9a64-e1e21e2094f9.png">

**Expected behavior**

Only one section is highlighted.

**Comment by guyius** [comment]
Hey @sapegin !
I started looking into this issue and found the root cause, Do you mind if I work on this?
It's going to be a first-time open source PR for me.

**Comment by sapegin** [comment]
That would be awesome! Let me know if you need any help :-)

**Comment by guyius** [comment]
Okay so the issue lies in the fact that the ComponentsListRenderer sets the isSelectedItem const according to the hasHash function, which checks for indexof, Thus Nav returns true for routes like NavLink. I reproduced it by changing CounterButton to ButtonCounter for example and colliding with Button. 
IMO we can simplify this and check in ComponentsListRenderer for equality by comparing the 2 strings instead of doing this: ```const isItemSelected = hasInHash(windowHash, href);```
We can simply do this: ```const isItemSelected = windowHash === href;```
Questions: 1. I prefer this simple logic to live in the ComponentsListRenderer seems to make sense for me, WDYT?
2. All tests pass with this change, and I saw that all the tests for ComponentsListRenderer are snapshots, Which also missed this edge case in the first place. Should I add specific tests for this edge case or leave this as is with the snapshot tests only?
